### ISSUE TITLE
[Bug] demo/redis_cache_demo.go is invalid syntax

### ISSUE THREAD

**Issue by huantt** [growthbook/growthbook-golang#19]
### Summary

[demo/redis_cache_demo.go](https://github.com/growthbook/growthbook-golang/blob/main/demo/redis_cache_demo.go) contains invalid syntax:

- The usage of `logError` is problematic because it's a private function in the `growthbook` package, and you cannot call it from the `demo` package.
- The line `growthbook.ConfigureCache(NewRedisFeatureCache("gb:", options))` at the end of the file is also invalid syntax.

#### Expected Behavior

I would like to propose the removal of this package for several reasons:

- The code contains invalid syntax.
- It does not seem relevant to this project and does not import the `growthbook` module.
- It imports `github.com/go-redis/redis/v8`, which is not relevant to the `growthbook` repository. This unnecessary import make  every user importing `growthbook` will need to download `github.com/go-redis/redis/v8` as well.

**Comment by ian-ross** [comment]
Yeah, this probably shouldn't be in here. It's a rough demonstration of how to use Redis for feature caching. It's used in a pending update to the documentation for the Go SDK (https://github.com/growthbook/growthbook/pull/1420), but it doesn't really belong here.
